receiver/k8sobjectsreceiver/config.go

```diff
@@ -50,14 +60,26 @@ type K8sObjectsConfig struct {
 type Config struct {
 	k8sconfig.APIConfig `mapstructure:",squash"`
 
-	Objects []*K8sObjectsConfig `mapstructure:"objects"`
+	Objects   []*K8sObjectsConfig `mapstructure:"objects"`
+	ErrorMode ErrorMode           `mapstructure:"error_mode"`
+	logger    *zap.Logger
 
 	// For mocking purposes only.
 	makeDiscoveryClient func() (discovery.ServerResourcesInterface, error)
 	makeDynamicClient   func() (dynamic.Interface, error)
 }
 
+func (c *Config) SetLogger(logger *zap.Logger) {
```

Comment:
Is this required? It looks weird to me that we need to set logger as part of the Config but maybe I miss sth here.